onesignal.php

```diff
@@ -39,6 +39,12 @@
 
     // Ensure migration is marked as complete after loading V3
     if (!$is_migrated) {
+        if(!$is_new_install) {
+            // Upgrade within v3 - enable notification_on_page for existing users
+            // This prevents breaking changes for users who were used to v2 behavior
+            $settings['notification_on_page'] = 1;
+            update_option('OneSignalWPSetting', $settings, 'no');
+        }
```

Comment:
I believe this is still going to only run once, when going from v2 to v3, since `is_migrated` is only `false` in that case.

Can we check both `is_new_install` and  if `notification_on_page` is present / non-null to determine if we need set a value in `OneSignalWPSetting`?

Reply:
Yep, good idea.